HW2/P3/P3.txt

```diff
@@ -0,0 +1,25 @@
+Do NOT use instruction-level parallelism:
+
+num_threads = 1
+1074.656613 Million Complex FMAs in 5.6932133201 seconds, 200.3313128311 million Complex FMAs / second
+
+num_threads = 2
+1074.656613 Million Complex FMAs in 2.7045673838 seconds, 433.6757743913 million Complex FMAs / second
+
+num_threads = 4
+1074.656613 Million Complex FMAs in 1.2576563890 seconds, 801.1923864326 million Complex FMAs / second
+
+
+Use instruction-level parallelism:
+
+num_threads = 1
+1074.94829 Million Complex FMAs in 10.9860551357 seconds, 97.8466134313 million Complex FMAs / second
+
+num_threads = 2
+1074.94829 Million Complex FMAs in 5.55642104149 seconds, 193.46055347 million Complex FMAs / second
+
+num_threads = 4
+1074.94829 Million Complex FMAs in 5.89242315292 seconds, 182.428902695 million Complex FMAs / second
+
+
+We can see from two approaches, using instruction-level parallelism and do NOT use instruction-level parallelism, the running time in the using instruction-level parallelism is longer than the do NOT use instruction-level parallelism. 
```

Comment:
I also got that AVX was slower. I remember Ray mentioning that some machines have to simulate AVX, and I figured that this was the reason for this counterintuitive result.